### PR TITLE
Ci/hunter release checkout all

### PR DESCRIPTION
**Pull request by hyperwidget**
Updates Deploy to S3 action to take an `is-canary` boolean flag; this then allows us to keep multiple canary builds and choose which one to deploy to sandbox.

This requires [the companion PR in the react-app-starter-kit](https://github.com/janeapp/react-app-starter-kit/pull/34) to be able properly serve canary builds using react-app-starter-kit apps

### CODE REVIEW

File: .github/workflows/release-frontend-app.yml [hunk 1]
```diff
@@ -1,51 +1,53 @@
...
-        output-folder:
-          description: 'Folder where the built assets are located. Also used to name the archive.'
-          default: 'build'
-          required: false
-          type: string
-        release-command:
-          description: 'Command used to release.'
-          default: 'yarn release'
-          required: false
-          type: string
```
Comment (hyperwidget) [inline .github/workflows/release-frontend-app.yml:39]: All auto formatting above

File: .github/workflows/release-frontend-app.yml [hunk 2]
```diff
@@ -1,51 +1,53 @@
...
   release:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
 
       - name: Prepare repository
-        run: git fetch --unshallow --tags
+        run: git fetch --tags
```
Comment (hyperwidget) [inline .github/workflows/release-frontend-app.yml:50]: above changes from Val's previous work

File: .github/workflows/deploy-frontend-app.yml [hunk 3]
```diff
@@ -16,10 +16,15 @@ on:
         required: false
         type: string
       node-version:
-        description: 'Node version to run with'
-        default: '16.x'
+        description: "Node version to run with"
+        default: "16.x"
         required: false
         type: string
+      is-canary:
```
Comment (hyperwidget) [inline .github/workflows/deploy-frontend-app.yml:23]: Added is canary flag, defaulting to false

File: .github/workflows/deploy-frontend-app.yml [hunk 4]
```diff
@@ -61,12 +66,24 @@ jobs:
...
+      - name: download canary assets
+        if: ${{ inputs.is-canary }}
+        uses: robinraju/release-downloader@v1.4
+        with:
+          tag: "v0.0.0-canary"
+          fileName: "build${{inputs.version_number}}.zip"
+      - name: decompress canary ZIP file
+        if: ${{ inputs.is-canary }}
+        run: |
+          unzip build${{inputs.version_number}}.zip
```
Comment (hyperwidget) [inline .github/workflows/deploy-frontend-app.yml:78]: custom logic for fetching and unpacking canary

Comment (carvallegro) [reply .github/workflows/deploy-frontend-app.yml:78]: Very small nitpick but how do you feel about adding a comment to explicitely indicate and document the branching flow between canary and non canary?

**Review by carvallegro — CHANGES_REQUESTED** [review]

**Review by carvallegro — APPROVED** [review]

**Pull request merged**